plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -309,13 +309,48 @@ public synchronized FakerOutputTableHandle beginCreateTable(ConnectorSession ses
             if (generator != null && !isCharacterColumn(column)) {
                 throw new TrinoException(INVALID_COLUMN_PROPERTY, "The `generator` property can only be set for CHAR, VARCHAR or VARBINARY columns");
             }
+            // only parse min, max, and options to validate literals - FakerColumnHandle needs to be serializable,
+            // and some internal Trino types are not (Int128, LongTimestamp, LongTimestampWithTimeZone), so they cannot be stored in the handle as native types
+            String min = (String) column.getProperties().get(ColumnInfo.MIN_PROPERTY);
+            try {
+                Literal.parse(min, column.getType());
+            }
+            catch (IllegalArgumentException e) {
+                throw new TrinoException(INVALID_COLUMN_PROPERTY, "The `min` property must be a valid " + column.getType().getDisplayName() + " literal", e);
+            }
+            String max = (String) column.getProperties().get(ColumnInfo.MAX_PROPERTY);
+            try {
+                Literal.parse(max, column.getType());
+            }
+            catch (IllegalArgumentException e) {
+                throw new TrinoException(INVALID_COLUMN_PROPERTY, "The `max` property must be a valid " + column.getType().getDisplayName() + " literal", e);
+            }
```

Comment:
nit: extract helper methods here. Specifically min/max handling is kinda identical. 
Also moving this catch driven logic outside will make caller method more readable

Reply:
I moved all of this to the `FakerColumnHandle` class, it looks much better, thanks!